qt/scientific_interfaces/Indirect/IndirectFitData.cpp

```diff
@@ -282,12 +282,17 @@ void IndirectFitData::validateSpectra(FunctionModelSpectra const &spectra) {
 
 void IndirectFitData::setStartX(double const &startX, WorkspaceIndex const &spectrum) {
   const auto range = m_ranges.find(spectrum);
-  if (range != m_ranges.end())
-    range->second.first = startX;
-  else if (m_workspace)
+  if (range != m_ranges.end()) {
+    if (range->second.second < startX) {
+      range->second.first = range->second.second;
```

Comment:
Can you tell me a bit more about these two lines? what are second.first and second.second in range?

Reply:
range here is an itterator for a map, `range.first` is a WorkspaceIndex, and `range.second` is a pair of doubles where the `range.second.first` is StartX and `range.second.second` is EndX